### PR TITLE
Solved DC #118 in linear time.

### PR DESCRIPTION
**Pull request by j-rewerts**
@CodyGramlich I bet you could make it slightly faster in certain edge cases.

I also use N extra space. It'd be possible to get rid of that by modifying the input list.

### CODE REVIEW

**Comment by CodyGramlich** [comment]
I had to refer to merge sort on this one. My solution also uses N extra space and is in linear time.

File: daily-coding/118/cody.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    # all numbers are positive
+    if numbers[i] >= 0:
+        for m in range(len(numbers)):
+            numbers[m] = numbers[m]**2
+        return numbers
+
+    # all numbers are negative
+    if numbers[j] < 0:
+        m = j
+        for n in range(len(numbers)//2):
```
Comment (j-rewerts) [inline daily-coding/118/cody.py]: How does this one work? How do you know you only need to do root(n) iterations?

Comment (CodyGramlich) [reply daily-coding/118/cody.py]: It iterates for half of numbers. n starts at the beginning and goes up and m starts at the end and goes down. I'm reversing the list while squaring them.

Comment (j-rewerts) [reply daily-coding/118/cody.py]: What does // do in this?

Comment (CodyGramlich) [reply daily-coding/118/cody.py]: I just realized that it wasn't squaring the middle number if the list has an odd number of elements, so I added a fix.

File: daily-coding/118/cody.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        t += 1
+
+    while s > len(positives):
+        numbers[t] = positives[s]
+        s += 1
+        t += 1
+
+    return numbers
+
+print(squareSort([-50, -20, -10]))
```
Comment (j-rewerts) [inline daily-coding/118/cody.py:66]: Could you uncomment these tests and add your new ones to my solution?

Comment (CodyGramlich) [reply daily-coding/118/cody.py:66]: Done.

**Pull request merged**